### PR TITLE
RD-939 - update(README, rock_eval_data) - add documentation

### PR DESCRIPTION
**Pull request by lorpacini**

### CODE REVIEW

**Comment by lorpacini** [comment]
Suggested modifications committed

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+
+[dev-packages]
+
```
Comment (parbelet) [inline Pipfile:9]:
```cfg
matplotlib = "3.5.0"
```

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+
```
Comment (parbelet) [inline Pipfile:7]:
```cfg
pandas = "1.1.3"
```

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,10 @@
...
 test_treated_data
-treat_data_rockeval_comparison.py
\ No newline at end of file
+treat_data_rockeval_comparison.py
+RD-974-treat_data_RE6_comparison.py
+RD-974-DATA/*
+RD-974-DATA_treated/*
+RD-998-treat_data_RE6_comparison.py
+RD-998-DATA/*
+RD-998-DATA_treated/*
```
Comment (parbelet) [inline .gitignore:12]: as the repo is public it'd be better to remove these

File: README.md [hunk 4]
```diff
@@ -1,3 +1,66 @@
...
+For the treatment, [pandas](https://pandas.pydata.org/) is required:
+```
+pip install pandas
+```
+
+## Example
+
+Requirement for plotting:
+```
+pip install matplotlib
```
Comment (parbelet) [inline README.md:23]: No need to specify the required packages here, the Pipfile is sufficient

File: README.md [hunk 5]
```diff
@@ -1,3 +1,66 @@
...
+    axs[0][i].grid(True)
+for j, f in enumerate(["T", "CO", "CO2"]):
+    axs[1][j].plot(curves["oxidation"]["time"], curves["oxidation"][f])
+    axs[1][j].set_xlabel("time")
+    axs[1][j].set_ylabel(f)
+    axs[1][j].set_title(f + " Oxidation")
+    axs[1][j].grid(True)
+plt.tight_layout()
+plt.show()
+```
```
Comment (parbelet) [inline README.md:66]: I'd rather move this in a separate script file (for instance `example.py`)

File: rock_eval_data.py [hunk 6]
```diff
@@ -169,9 +208,23 @@ def normalize_gas(x, mass_ratio):
         return {"pyrolysis": curves_pyr, "oxidation": curves_oxi}
         
     def get_metadata(self):
```
Comment (parbelet) [inline rock_eval_data.py:210]: Thanks for the typing!

You can also specify the return type
For instance here:

```python
def get_metadata(self) -> Dict:
```

File: rock_eval_data.py [hunk 7]
```diff
@@ -106,7 +137,15 @@ def _extract_curves(self):
         
         return {"pyrolysis": curves_pyr.astype(float), "oxidation": curves_oxi.astype(float)}
 
-    def _normalize_curves(self, raw_curves):
+    def _normalize_curves(self, raw_curves: dict):
```
Comment (parbelet) [inline rock_eval_data.py:140]: Here, use 
```python
from typing import Dict

def _normalize_curves(self, raw_curves: Dict) -> Dict:
```

File: Pipfile (hunk 1 above)
Comment (lorpacini) [reply Pipfile:9]: This dependency is necessary for the example but not for the processing. Should I put it anyway? Or should I remove the plotting from the example?

**Pull request merged**